openmc/filter.py

```diff
@@ -0,0 +1,1810 @@
+            raise ValueError("Expected HDF5 data for filter type '"
+                             + cls.short_name.lower() + "' but got '"
+                             + group['type'][()].decode() + " instead")
+
+        filter_id = int(group.name.split('/')[-1].lstrip('filter '))
+
+        out = cls(group['bins'][()], filter_id=filter_id)
+        out._num_bins = group['n_bins'][()]
+
+        return out
+
+    @property
+    def num_bins(self):
+        # Need to handle number of bins carefully -- for distribcell tallies, we
+        # need to know how many instances of the cell there are
+        return self._num_bins
+
+    @property
+    def paths(self):
+        return self._paths
+
+    @Filter.bins.setter
+    def bins(self, bins):
+        # Format the bins as a 1D numpy array.
+        bins = np.atleast_1d(bins)
+
+        # Make sure there is only 1 bin.
+        if not len(bins) == 1:
+            msg = 'Unable to add bins "{0}" to a DistribcellFilter since ' \
+                  'only a single distribcell can be used per tally'.format(bins)
+            raise ValueError(msg)
+
+        # Check the type and extract the id, if necessary.
+        cv.check_type('distribcell bin', bins[0], (Integral, openmc.Cell))
+        if isinstance(bins[0], openmc.Cell):
+            bins = np.atleast_1d(bins[0].id)
+
+        self._bins = bins
+
+    @paths.setter
+    def paths(self, paths):
+        cv.check_iterable_type('paths', paths, str)
+        self._paths = paths
+
+    def can_merge(self, other):
+        # Distribcell filters cannot have more than one bin
+        return False
+
+    def get_bin_index(self, filter_bin):
+        # Filter bins for distribcells are indices of each unique placement of
+        # the Cell in the Geometry (consecutive integers starting at 0).
+        return filter_bin
+
+    def get_pandas_dataframe(self, data_size, stride, **kwargs):
+        """Builds a Pandas DataFrame for the Filter's bins.
+
+        This method constructs a Pandas DataFrame object for the filter with
+        columns annotated by filter bin information. This is a helper method for
+        :meth:`Tally.get_pandas_dataframe`.
+
+        Parameters
+        ----------
+        data_size : int
+            The total number of bins in the tally corresponding to this filter
+        stride : int
+            Stride in memory for the filter
+
+        Keyword arguments
+        -----------------
+        paths : bool
+            If True (default), expand distribcell indices into multi-index
+            columns describing the path to that distribcell through the CSG
+            tree.  NOTE: This option assumes that all distribcell paths are of
+            the same length and do not have the same universes and cells but
+            different lattice cell indices.
+
+        Returns
+        -------
+        pandas.DataFrame
+            A Pandas DataFrame with columns describing distributed cells.  The
```

Comment:
This sentence seems incomplete `The for will be either:`. I am not clear what the correct substitution is though.